### PR TITLE
[WIP] new(docker): introduce a new development docker image

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area build

**What this PR does / why we need it**:

This PR adds a new docker image for testing Falco during the development

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(docker): introduce a new development docker image
```

### CODE REVIEW

**Comment by Andreagit97** [comment]
This requires https://github.com/falcosecurity/libs/pull/679

File: docker/dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    rm -rf /falco/usr/src; \
+    rm /falco/usr/bin/falco-driver-loader
+
+# the time displayed in log messages and output messages will be in ISO 8601.
+RUN sed -e 's/time_format_iso_8601: false/time_format_iso_8601: true/' < /falco/etc/falco/falco.yaml > /falco/etc/falco/falco.yaml.new; \
+    mv /falco/etc/falco/falco.yaml.new /falco/etc/falco/falco.yaml
+
+# Please note: it could be necessary to change this base image according
+# to the `glibc` version of the machine where you build the tar.gz package
+FROM ubuntu:22.04 AS runner
```
Comment (Andreagit97) [inline docker/dev/Dockerfile:21]: I need to rebase the PR on master and allow the user to customize this `ubuntu:22.04` with a docker variable

**Comment by Andreagit97** [comment]
/hold
until https://github.com/falcosecurity/falco/pull/2302 is merged, we need bundled libelf

**Comment by leogr** [comment]
Hey @Andreagit97 

Could you rebase this pls? :pray:

**Pull request closed without merging**